### DISCUSSION TITLE
legacy help

### DISCUSSION THREAD

**Discussion by alexanderpowerplays** [ALTaleX531/OpenGlass#70] [Q&A]
i was wondering how to set the legacy branch to the windows 7 style rather than the windows 10 style since i cant seem to figure out how to change it

**Comment by DAPOPOBEFASTONYOAZZ** [comment]
Need more information...
Please provide more information about your desired goal so we can help you.

**Comment by alexanderpowerplays** [comment]
when I install it, it makes the windows 10 ui become see through

but I'm trying to figure out how to make it look like windows 7's frutiger aero like in this picture ![Aero_Example.png](https://github.com/user-attachments/assets/ea3f8e6d-7f92-4cfc-81f4-9a0550ef6c80)

**Comment by DAPOPOBEFASTONYOAZZ** [comment]
Installing OpenGlass does not automatically install any Windows Aero themes, all it does is re-implement functions in the code to draw blur in window borders again.

If you want to get a Windows Aero theme there are several steps you must take. You will first need to install a theme patcher. I use UltraUXThemePatcher.

Next, you'll need to download a Windows Aero theme. One of the best and most accurate to use is: https://www.deviantart.com/vaporvance/art/Aero10-for-Windows-10-1903-22H2-909711949.

You'll need to know how to install a theme into Windows from there.

If you want to change your colors or other settings, I recommend using WinPaletter to begin, but you also need to be familiar with editing the registry. I've attached my WinPaletter themes you need to first apply for the default color choices included in Windows Vista and Windows 7 - it uses Windows 8's default color as a fallback for any Windows 10 menus, but you can change this if you'd like. You can either choose from one of the standard color choices, or you can choose one of the "Transparency Disabled" choices. "Transparency Disabled" choices will adjust the colors to look exactly as if you turned off the transparency setting in the Windows Vista or Windows 7 control panel. This is *not* the same effect as Windows Vista when you maximize a window and the maximized window becomes opaque and darkens the color, it is only accurate to the Windows Vista colors when referring to any non-maximized windows.

After applying a theme, then you'll need to use the registry editor to further tweak settings to give it an accurate appearance. I also included my reference that shows each key's hex value that you'll need to adjust to get the accurate effect of each default color choice that you choose. A few notes:
- For GlassType, if you want to use a Windows 7 style color, use GlassType = 1 for color multiplication. For Windows Vista (or any of the transparency disabled) colors, use GlassType = 0. This only includes the color tint + blur, no color multiplication blending mode.
- The BlurDeviation registry key must be set to the hex value of "A" (or 10 in decimal value) for the accurate blur effect used in Windows Vista and Windows 7.
- If there are any registry keys mentioned in the text file that you do not already have in your registry located at Computer\HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\DWM, you will need to create them. When creating a regsitry key, all registry keys must DWORDs.
- If you are using a "Transparency Disabled" color, to get the best and most accurate effect, you will need to set your GlassOpacity registry key hex value to "64" (or 100 in decimal.)


You will also need StartIsBack to restore the classic taskbar with the Windows 7 taskbar, start menu and start button as well if you have not already. You need to make any color adjustments to the taskbar through StartIsBack after applying a WinPaletter theme. These settings are also included in my text reference file as the RGBA settings, A being the transparency value for your start menu and taskbar.

To tie it together, you will need to install AeroWindowManager, or AWM, to adjust the size of the window control buttons. You can download it from another Github repository. Installation instructions are included on the Github page. However, settings for Windows Vista or Windows 7 sized buttons are not included. I also included my reference file that gives an accurate representation of the button sizes from Windows Vista, which buttons are slightly smaller, to Windows 7, which buttons are slightly bigger than Windows Vista. It shows the values that you need to change to get the look you want.

**Registry colors:** 
[registry colors.txt](https://github.com/user-attachments/files/17922324/registry.colors.txt)
**Registry colors (transparency disabled):**
[registry colors - transparency disabled.txt](https://github.com/user-attachments/files/17922351/registry.colors.-.transparency.disabled.txt)
**WinPaletter themes (these files are compatible with version 1.0.9.3):**
[WinPaletter Themes.zip](https://github.com/user-attachments/files/17922375/WinPaletter.Themes.zip)
**AWM Reference File:**
[AWM Settings.txt](https://github.com/user-attachments/files/17922398/AWM.Settings.txt)

**Comment by alexanderpowerplays** [comment]
ah, thank you for the detailed explanation 

if I need help or have any questions, I'll go ahead and leave a comment with anything I need help with or any questions I have

**Comment by DAPOPOBEFASTONYOAZZ** [comment]
You're welcome and sounds good! If I resolved your question, please be sure to mark this case as closed since it is solved.
